server/controllers/refer/courseParticipationDetailsController.test.ts

```diff
@@ -0,0 +1,79 @@
+import { faker } from '@faker-js/faker'
+import type { DeepMocked } from '@golevelup/ts-jest'
+import { createMock } from '@golevelup/ts-jest'
+import type { NextFunction, Request, Response } from 'express'
+import createError from 'http-errors'
+
+import CourseParticipationDetailsController from './courseParticipationDetailsController'
+import type { CourseService, PersonService, ReferralService } from '../../services'
+import { courseParticipationFactory, personFactory, referralFactory } from '../../testutils/factories'
+
+describe('CourseParticipationDetailsController', () => {
+  const token = 'SOME_TOKEN'
+  let request: DeepMocked<Request>
+  let response: DeepMocked<Response>
+  const next: DeepMocked<NextFunction> = createMock<NextFunction>({})
+
+  const courseService = createMock<CourseService>({})
+  const personService = createMock<PersonService>({})
+  const referralService = createMock<ReferralService>({})
+
+  let courseParticipationDetailsController: CourseParticipationDetailsController
+
+  beforeEach(() => {
+    request = createMock<Request>({ user: { token } })
+    response = createMock<Response>({})
```

Comment:
Now that I've merged in that change to the prisoner search endpoint, you'll need to add a user to this to pass caseloads to `getPerson`. You can use `createMockResponseWithCaseloads` in the newly-created `testutils/helpers.ts`.